src/object/cli_obj.c

```diff
@@ -358,7 +362,15 @@ obj_layout_create(struct dc_object *obj, unsigned int mode, bool refresh)
 		obj_shard->do_reintegrating = layout->ol_shards[i].po_reintegrating;
 		obj_shard->do_target_rank = layout->ol_shards[i].po_rank;
 		obj_shard->do_target_idx = layout->ol_shards[i].po_index;
+		if (obj->cob_min_rank > obj_shard->do_target_rank)
+			obj->cob_min_rank = obj_shard->do_target_rank;
+		if (obj->cob_max_rank < obj_shard->do_target_rank)
+			obj->cob_max_rank = obj_shard->do_target_rank;
 	}
+
+	if (unlikely(obj->cob_min_rank == -1))
+		obj->cob_min_rank = 0;
```

Comment:
Yes, it will avoid unnecessary overhead. The "cob_rank_nr" will be only be set after parsing the whole object layout via obj_coll_oper_args_collapse() since collective operations are the users.